jscomp/bsb/bsb_ninja_rule.ml

```diff
@@ -164,11 +164,11 @@ let make_custom_rules ~(gentype_config : Bsb_config_types.gentype_config option)
     | None -> ()
     | Some flag ->
         Ext_buffer.add_char_string buf ' ' (Bsb_build_util.pp_flag flag));
-    (match (has_reason_react_jsx, reason_react_jsx, jsx_version) with
-    | _, _, Some Jsx_v3 -> Ext_buffer.add_string buf " -bs-jsx 3"
-    | _, _, Some Jsx_v4 -> Ext_buffer.add_string buf " -bs-jsx 4"
-    | _, Some Jsx_v3, None -> Ext_buffer.add_string buf " -bs-jsx 3"
-    | _ -> ());
+    (match (reason_react_jsx, jsx_version) with
+    | _, Some Jsx_v3 -> Ext_buffer.add_string buf " -bs-jsx 3"
+    | _, Some Jsx_v4 -> Ext_buffer.add_string buf " -bs-jsx 4"
+    | Some Jsx_v3, None -> Ext_buffer.add_string buf " -bs-jsx 3"
+    | None, None -> ());
```

Comment:
what does this case do currently?

wondering about backwards compatibility for existing projects, as the default should be behave like the old compiler

Reply:
Yes, the 4th branch is for backward compatibility in case the current config `reason.react-jsx` and the new config `jsx.version` both are not existing. Additionally, the 3rd branch is also for backward compatibility in case that `reason.react-jsx` config is existing